### ISSUE TITLE
Signature Example

### ISSUE THREAD

**Issue by rchadwic** [adlnet/xAPI-Spec#905]
I think the example of a signature attachment is incorrect. The base64 provided, when decoded, is just the header portion of the token. I think the example should be the base64 encoding of the whole token, something like

```
base64( base64(token_header_json) +"." + base64(token_payload) + "." + base64(hmac_sig))
```

**Comment by garemoko** [comment]
If the example is wrong, this should probably be a patch issue rather than a minor issue.

**Comment by ljwolford** [comment]
After looking at the example again I thought of something else to add to this...Is the example creating the signature without the 'attachments' field then just adding it onto the statement later? If not, how does the sha2 field get validated since the sha2 value should be made from the signature (which is made partially from the statement)? If it is creating the signature based off the statement without the attachments field then that should be listed in the Signed Statements section under the equivalence check.

**Comment by garemoko** [comment]
@jljwoldford There's already a clear requirement in that section:

"The JWS signature MUST have a payload of a valid JSON serialization of the Statement generated before the signature was added."

I guess we could add "minus the attachment" to "Validate that the "original" Statement is logically equivalent to the received Statement."

**Comment by oconnetf** [comment]
Hmm. I suspect we might've discovered a common copy/paste issue. If I double-click the "JSW signature" text in my browser (Safari in this case) and paste it, I can reproduce the lack of payload. But that's because the built-in word detection doesn't capture the entire signature.

The signature line is ​_really_​ long, and it's '.' delimited. So if you double-click it you get a string ending in "….Q0d8" which is only the header. 

If I'm careful to copy/paste the entire signature, I get the payload I expect. I think I recommend closing this issue.

As for @ljwolford's suggestion, I conferred with @bscSCORM, who suggested we have to verify “except for the signature itself” since the attachment will of course not be there. If there's something that could add clarity here, that's fine. But I think we've validated that the core concern of this issue is not itself a problem.
